lib/build/l10n/makePackages.js

```diff
@@ -1,33 +1,226 @@
+    }
+
+  fconsole.endl();
+
+  //
+  // inject package load script into module code
+  //
+
+  var injection = {
+    body: function(cultureList, defaultCulture, defaultPackage, packageDigest, storagePrefix){
+      /*global basis, currentCulture, setCulture, setCultureList, onCultureChange, Dictionary, dictionaryByUrl, internalResolveDictionary */
+
+      setCultureList(cultureList);
+      setCulture(defaultCulture);
+
+      var packages = {};
+      packages[defaultCulture] = [defaultPackage];
+
+      basis.resource.extensions['.l10n'] = function(content, url){
+        if (getPackage(currentCulture))
+        {
+          var dictOffset = parseInt(url.match(/.+\/(.+)\.l10n$/)[1], 36);
+          var dictContent = {};
+          dictContent[currentCulture] = packages[currentCulture][0][dictOffset];
+          return internalResolveDictionary(url, true).update(dictContent);
+        }
+      };
+
+      Dictionary.extend(function(super_, proto_){
+        return {
+          syncValues: function(){
+            if (getPackage(currentCulture))
+              proto_.syncValues.call(this);
+          }
+        };
+      });
+
+      onCultureChange(function(culture){
+        var data = getPackage(culture);
+        if (data)
+          applyPackage(data);
+        else
+          loadPackage(culture);
+      }, null, true);
+
+      function getPackage(culture){
+        var data = packages[culture];
+        if (!data)
+          try { data = packages[culture] = basis.json.parse(global.localStorage[storagePrefix + ':' + culture]); } catch(e) {};
+
+        return data && (culture == defaultCulture || data[1] == packageDigest[culture]) && data[0];
+      }
+
+      function applyPackage(data){
+        for (var i = 0, patch; patch = data[i]; i++)
+        {
+          var url = basis.path.resolve(i.toString(36) + '.l10n');
+          if (dictionaryByUrl[url])
+          {
+            var dictContent = {};
+            dictContent[currentCulture] = patch;
+            dictionaryByUrl[url].update(dictContent);
+          }
+        }
+      }
+
+      function loadPackage(culture){
+        var xhr = new global.XMLHttpRequest();
+
+        xhr.onreadystatechange = function(){
+          if (this.readyState == 4 && ((this.status >= 200 && this.status < 300) || this.status == 304))
+          {
+            var data;
+            try { data = basis.json.parse(this.responseText); } catch(e) {};
+            if (data)
+            {
+              packages[culture] = [data, packageDigest[culture]];
+              global.localStorage[storagePrefix + ':' + culture] = JSON.stringify(packages[culture]);
+              applyPackage(data);
+            }
+          }
+        };
+
+        xhr.open('GET', './l10n/' + culture + '.culture?' + packageDigest[culture], true);
+        xhr.send();
+      }
+    },
+
+    args: [
+      cultureList,
+      defaultCulture,
+      packages[defaultCulture],
+      packageDigest,
```

Comment:
May be better `culturePackages` instead of `packageDigest`?

```js
culturePackages = {
  'ru-RU': { cacheKey: 'l10n:ru-RU', digest: '...', ... }
}
```